sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -55,7 +55,7 @@ def _get_validated_statsforecast_params(self):
         sktime_default_params = self.get_param_defaults().keys()
         statsforecast_params = self._get_init_statsforecast_params()
 
-        for sktime_param in sktime_params.keys():
+        for sktime_param in list(sktime_params):
             if sktime_param not in statsforecast_params:
                 sktime_params.pop(sktime_param)
```

Comment:
hm, could we perhaps avoid mutating operations overall?

Reply:
In this case is necessary, the original problem came from this:

```python
d = {"a":1}
for key in d.keys():
    d.pop(key)

# Will throw: RuntimeError: dictionary changed size during iteration

# Instead try this:

d = {"a":1}
for key in list(d):
    d.pop(key)

# Will do what we expect.
```

But i can try some examples in .py file with version 1.6.0 before/after if you want

Reply:
why not this: `statsforecast_params = {k: v for k, v in sktime_params.items()}`

let´s just avoid `pop` altogether

Reply:
I do not really see a problem with `pop`, thats how it is right now:
```python
# Version 1
for sktime_param in list(sktime_params):
    if sktime_param not in statsforecast_params:
        sktime_params.pop(sktime_param)
        if sktime_param not in sktime_default_params:
            warn(
                f"Keyword argument '{sktime_param}' will be omitted as it is"
                f" not found in the __init__ method "
                f"from {self._get_statsforecast_class()}. "
                f"Check your statsforecast version"
                f"to find out the right API parameters."
            )

```
You think this would be cleaner?

```python
# Version 2
sktime_params = self._get_statsforecast_params()
sktime_default_params = self.get_param_defaults().keys()
statsforecast_params = self._get_init_statsforecast_params()

filtered_sktime_params = {
    key: value for key, value in sktime_params.items() if key in statsforecast_params
}

for sktime_param in sktime_params:
    if sktime_param not in statsforecast_params:
        if sktime_param not in sktime_default_params:
            warn(
                f"Keyword argument '{sktime_param}' will be omitted as it is"
                f" not found in the __init__ method "
                f"from {self._get_statsforecast_class()}. "
                f"Check your statsforecast version"
                f"to find out the right API parameters."
            )
```

I actually like more the first version

Reply:
I also will prefer to not have in place modifications with pop. And instead of a for loop over all keys, I think you can do a set difference and throw a single warning for all problematic ones.

Reply:
What about this:

```python
sktime_params = self._get_statsforecast_params()
sktime_default_params = set(self.get_param_defaults().keys())
statsforecast_params = set(self._get_init_statsforecast_params())

# Filter sktime_params to only include keys in statsforecast_params
filtered_sktime_params = {key: value for key, value in sktime_params.items() if key in statsforecast_params}

# Find parameters not in statsforecast_params or sktime_default_params
param_diff = sktime_params.keys() - statsforecast_params - sktime_default_params

for sktime_param in param_diff:
    warn(
        f"Keyword argument '{sktime_param}' will be omitted as it is"
        f" not found in the __init__ method "
        f"from {self._get_statsforecast_class()}. "
        f"Check your statsforecast version"
        f"to find out the right API parameters."
    )


```

Reply:
I really don't follow the logic to throw multiple warnings in a loop for every single one, one seems sufficient enough. Other than, this looks okay to me.

Reply:
Yeah you are totally right, i just did the warnings in a for loop because the other logic was also in for loops.